programs/gummyroll/src/lib.rs

```diff
@@ -44,6 +45,22 @@ pub struct Append<'info> {
     pub append_authority: Signer<'info>,
 }
 
+#[derive(Accounts)]
+pub struct VerifyLeaf<'info> {
+    /// CHECK: This account is validated in the instruction
+    pub merkle_roll: UncheckedAccount<'info>,
+}
+
+#[derive(Accounts)]
+pub struct TransferAuthority<'info> {
+    #[account(mut)]
+    /// CHECK: This account is validated in the instruction
+    pub merkle_roll: UncheckedAccount<'info>,
+    pub authority: Signer<'info>,
```

Comment:
What's the expected workflow for changing the `append_authority`? Higher level contract will need to do some sort of authorization. Does it make sense to have the `append_authority` also sign here?

Reply:
Workflow is that the `transferAuthority` instruction allows `authority` to change either one or both of `authority` and `appendAuthority`.

The transfer authority takes in `authority: Option<Pubkey>, append_authority: Option<Pubkey>`

Reply:
So after a `GumballMachine` is fully minted out. How does `willy_wonka` get removed as the `append_authority` of the `Gummyroll`?